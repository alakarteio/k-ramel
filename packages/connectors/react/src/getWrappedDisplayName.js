export default Component => (
  Component.displayName
  || Component.name
  || (Component.constructor && Component.constructor.name)
  || 'Unknown'
)
