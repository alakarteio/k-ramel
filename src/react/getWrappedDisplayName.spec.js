/* eslint-env jest */
import getWrappedDisplayName from './getWrappedDisplayName'

describe('react/getWrappedDisplayName', () => {
  it('should print a displayName', () => {
    expect({
      'Component.displayName': getWrappedDisplayName({ displayName: 'Component.displayName' }),
      'Component.name': getWrappedDisplayName({ name: 'Component.name' }),
      'Component.constructor.name': getWrappedDisplayName({ constructor: { name: 'Component.constructor.name' } }),
      Unknown: getWrappedDisplayName({ constructor: {} }),
    }).toMatchSnapshot()
  })
})
