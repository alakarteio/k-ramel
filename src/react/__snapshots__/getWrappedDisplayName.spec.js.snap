// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/getWrappedDisplayName should print a displayName 1`] = `
Object {
  "Component.constructor.name": "Component.constructor.name",
  "Component.displayName": "Component.displayName",
  "Component.name": "Component.name",
  "Unknown": "Unknown",
}
`;
